rust/lance/src/dataset.rs

```diff
@@ -2590,6 +2630,143 @@ mod tests {
         assert!(matches!(result.unwrap_err(), Error::DatasetNotFound { .. }));
     }
 
+    #[tokio::test]
+    async fn test_drop() {
+        let schema = Arc::new(ArrowSchema::new(vec![
+            Field::new("i", DataType::Int32, false),
+            Field::new("x", DataType::Float32, false),
+        ]));
```

Comment:
Could we test that this works when deleting a nested column (one in a struct?)

Reply:
Added a test case.